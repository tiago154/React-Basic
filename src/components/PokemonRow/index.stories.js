import React from 'react'
import PokemonRow from '.'
import { withKnobs, text, number } from '@storybook/addon-knobs'

export default {
  title: 'PokemonRow',
  decorators: [withKnobs]
}

export const Linha = () =>
  <PokemonRow
    pokemon={
      {
        id: number('id', 7),
        name: text('name', 'Squirtle'),
        image: text('image', 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/7.png')
      }
    }
  />
