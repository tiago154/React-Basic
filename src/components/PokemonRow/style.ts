import styled from 'styled-components'

export const Div = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 0px;
  width: 360px;
  height: 32px;
  align-items: center;
  background: rgba(94, 174, 169, 0.19);
`

export const Sprite = styled.img`
  width: 32px;
  height: 32px;
`

export const Text = styled.p`
  font-family: 'Press Start 2P', cursive;
  display: flex;
  color: #000;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
`
