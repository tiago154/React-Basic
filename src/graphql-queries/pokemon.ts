import { gql } from 'apollo-boost'
import { VIEWING_LIMIT } from '../constant'

export const LIST_POKEMONS_QUERY = (offset: number = 0, limit: number = VIEWING_LIMIT) => gql`
    {
        listPokemon (offset: ${offset} limit: ${limit})  {
            count pokemons {
                id name image types
            }
        }
    }
`