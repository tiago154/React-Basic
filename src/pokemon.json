[
    {
        "name": "Bulbasaur",
        "image": "https://assets.pokemon.com/assets/cms2/img/pokedex/full/001.png"
    },
    {
        "name": "Charmander",
        "image": "https://assets.pokemon.com/assets/cms2/img/pokedex/full/004.png"
    },
    {
        "name": "Squirtle",
        "image": "https://assets.pokemon.com/assets/cms2/img/pokedex/full/007.png"
    },
    {
        "name": "Chikorita",
        "image": "https://assets.pokemon.com/assets/cms2/img/pokedex/full/152.png"
    },
    {
        "name": "Cyndaquil",
        "image": "https://assets.pokemon.com/assets/cms2/img/pokedex/full/155.png"
    },
    {
        "name": "Totodile",
        "image": "https://assets.pokemon.com/assets/cms2/img/pokedex/full/158.png"
    },
    {
        "name": "Treecko",
        "image": "https://assets.pokemon.com/assets/cms2/img/pokedex/full/252.png"
    },
    {
        "name": "Torchic",
        "image": "https://assets.pokemon.com/assets/cms2/img/pokedex/full/255.png"
    },
    {
        "name": "Mudkip",
        "image": "https://assets.pokemon.com/assets/cms2/img/pokedex/full/258.png"
    },
    {
        "name": "Turtwig",
        "image": "https://assets.pokemon.com/assets/cms2/img/pokedex/full/387.png"
    },
    {
        "name": "Chimchar",
        "image": "https://assets.pokemon.com/assets/cms2/img/pokedex/full/390.png"
    },
    {
        "name": "Piplup",
        "image": "https://assets.pokemon.com/assets/cms2/img/pokedex/full/393.png"
    },
    {
        "name": "Snivy",
        "image": "https://assets.pokemon.com/assets/cms2/img/pokedex/full/495.png"
    },
    {
        "name": "Tepig",
        "image": "https://assets.pokemon.com/assets/cms2/img/pokedex/full/498.png"
    },
    {
        "name": "Oshawott",
        "image": "https://assets.pokemon.com/assets/cms2/img/pokedex/full/501.png"
    },
    {
        "name": "Chespin",
        "image": "https://assets.pokemon.com/assets/cms2/img/pokedex/full/650.png"
    },
    {
        "name": "Fennekin",
        "image": "https://assets.pokemon.com/assets/cms2/img/pokedex/full/653.png"
    },
    {
        "name": "Froakie",
        "image": "https://assets.pokemon.com/assets/cms2/img/pokedex/full/656.png"
    }
]