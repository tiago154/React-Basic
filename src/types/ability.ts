export type Ability = {
  name: string,
  description: string
}
