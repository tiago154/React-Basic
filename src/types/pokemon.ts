import { Weakness } from './weakness'

export type Pokemon = {
  id: number,
  name: string,
  image: string,
  types?: string[],
  weaknesses?: Weakness[]
}
