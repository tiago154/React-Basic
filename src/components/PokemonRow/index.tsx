import React, { useState } from 'react'
import { Div, Sprite, Text } from './style'
import { Pokemon } from '../../types/pokemon'
import { pokeballRolling } from '../../assets'

interface IProps {
  pokemon: Pokemon,
  onSelectPokemon: (pokemon: Pokemon) => void
}

const PokemonRow: React.FC<IProps> = ({ pokemon, onSelectPokemon }) => {
  const [imageIsLoaded, setLoadedImage] = useState(false)
  const { id, name, image } = pokemon
  return (
    <Div onClick={() => onSelectPokemon(pokemon)}>
      <Text>{id.toString().padStart(3, '0')} {name}</Text>
      <Sprite
        src={imageIsLoaded ? image : pokeballRolling}
        onLoad={() => setLoadedImage(true)}
      />
    </Div>
  )
}

export default PokemonRow
