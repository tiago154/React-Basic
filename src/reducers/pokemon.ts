import { Pokemon } from '../types/pokemon'

export enum PokemonActionsEnum {
  'LOAD_POKEMONS' = 'LOAD_POKEMONS',
  'ADD_MORE_POKEMONS' = 'ADD_MORE_POKEMONS',
  'SAME_STATE' = 'SAME_STATE',
  'UPDATE' = 'UPDATE'
}

export type PokemonState = {
  count: number,
  pokemons: Pokemon[],
  offSet: number
}

export interface PokemonsAction {
  type: PokemonActionsEnum
  payload: PokemonState
}

export type PokemonActionsTypes = PokemonsAction

const initialState: PokemonState = {
  count: 0,
  pokemons: [],
  offSet: 0
}

export default (state = initialState, action: PokemonActionsTypes): PokemonState => {
  switch (action.type) {
    case PokemonActionsEnum.LOAD_POKEMONS:
      return { ...state, ...action.payload }
    case PokemonActionsEnum.ADD_MORE_POKEMONS:
      return {
        ...action.payload,
        pokemons: [...state.pokemons, ...action.payload.pokemons]
      }
    case PokemonActionsEnum.UPDATE:
      return action.payload
    case PokemonActionsEnum.SAME_STATE:
      return action.payload
    default:
      return state
  }
}
