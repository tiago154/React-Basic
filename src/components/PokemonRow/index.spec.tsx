import React from 'react'
import renderer from 'react-test-renderer'
import PokemonRow from '.'
import { Pokemon } from '../../types/pokemon'

describe('PokemonType', () => {
  it('Renders correctly', () => {
    const fakePokemon: Pokemon = {
      id: 123,
      image: 'fakeImage',
      name: 'fakeName'
    }

    const tree = renderer
      .create(
        <PokemonRow
          pokemon={fakePokemon}
          onSelectPokemon={() => { }}
        />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
