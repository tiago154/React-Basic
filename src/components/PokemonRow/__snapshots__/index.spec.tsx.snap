// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonType Renders correctly 1`] = `
<div
  className="sc-bdVaJa cNuUBK"
  onClick={[Function]}
>
  <p
    className="sc-htpNat dhvPnF"
  >
    123
     
    pokemonFake
  </p>
  <img
    className="sc-bwzfXH dmdPBJ"
    onLoad={[Function]}
    src="default"
  />
</div>
`;
